datasources/element/element.go

```diff
@@ -0,0 +1,595 @@
+		cleanBody := msg.Content.Body
+		// Replace newlines with spaces or remove them
+		cleanBody = strings.ReplaceAll(cleanBody, "\n", " ")
+		cleanBody = strings.ReplaceAll(cleanBody, "\r", " ")
+		// Collapse multiple spaces
+		for strings.Contains(cleanBody, "  ") {
+			cleanBody = strings.ReplaceAll(cleanBody, "  ", " ")
+		}
+		cleanBody = strings.TrimSpace(cleanBody)
+
+		if cleanBody != msg.Content.Body && cleanBody != "" {
+			possibleFilenames = append(possibleFilenames,
+				fmt.Sprintf("%s/%s-%s", dir, cleanBody, timeStr))
+		}
+	}
+
+	// Also try the original filename if specified
+	if msg.Content.Filename != "" {
+		// Use appropriate directory based on content type
+		var dir string
+		switch msg.Content.MsgType {
+		case "m.image":
+			if strings.Contains(msg.Content.Filename, "sticker") || strings.Contains(msg.Content.URL, "sticker") {
+				dir = "stickers"
+			} else {
+				dir = "images"
+			}
+		case "m.video":
+			dir = "videos"
+		case "m.audio":
+			dir = "audio"
+		case "m.file":
+			dir = "files"
+		default:
+			dir = "images" // fallback
+		}
+		possibleFilenames = append(possibleFilenames,
+			fmt.Sprintf("%s/%s", dir, msg.Content.Filename))
+	}
+
+	// Try some variations with different extensions
+	for _, base := range possibleFilenames {
+		candidates := []string{base}
+		// Try without extension first, then with common extensions
+		if !strings.Contains(base, ".") {
+			switch msg.Content.MsgType {
+			case "m.image":
+				candidates = append(candidates, base+".jpg", base+".png", base+".jpeg")
+			case "m.video":
+				candidates = append(candidates, base+".mp4", base+".mov", base+".avi")
+			case "m.audio":
+				candidates = append(candidates, base+".ogg", base+".mp3", base+".wav")
+			default:
+				candidates = append(candidates, base+".jpg") // fallback
+			}
+		}
+
+		// Also try with dot and space variations (Element sometimes modifies special chars)
+		if strings.Contains(base, ".") && !strings.HasSuffix(base, ".jpg") && !strings.HasSuffix(base, ".png") && !strings.HasSuffix(base, ".mp4") && !strings.HasSuffix(base, ".ogg") {
+			spaceVersion := strings.ReplaceAll(base, ".", " ")
+			hyphenVersion := strings.ReplaceAll(base, ".", "-")
+			candidates = append(candidates, spaceVersion, hyphenVersion)
+		}
+
+		for _, filename := range candidates {
+			if fileExists(dirEntry.FS, filename) {
+				return createAttachmentItemWithFile(filename, timestamp, owner, msg, dirEntry.FS)
+			}
+		}
+	}
+
+	// If we can't find the file, create a placeholder attachment
+	return createAttachmentItemWithFile("", timestamp, owner, msg, dirEntry.FS)
+}
+
+// createAttachmentItemWithFile creates the actual attachment item
+func createAttachmentItemWithFile(filename string, timestamp time.Time, owner timeline.Entity, msg elementMessage, filesystem fs.FS) *timeline.Item {
+	var fileKey timeline.ItemRetrieval
+	fileKey.SetKey(fmt.Sprintf("element-%s-%s", timestamp.Format(time.DateTime), msg.EventID))
```

Comment:
Is this actually used/needed? Retrieval keys should only be set if the item can't be (efficiently) sent to the processor all at once. For example, in Google Photos, we use a retkey because sometimes we have the image, sometimes the metadata, but we don't have both at the same time, and can't reliably get both at the same time, so we use the retkey to link them together.

If you have the whole item all together, or can efficiently get the rest of its info, don't set a retrieval key.

Reply:
It wasn' needed - I removed it.